core/chains/evm/client/chain_client.go

```diff
@@ -272,10 +279,48 @@ func (c *chainClient) LatestFinalizedBlock(ctx context.Context) (*evmtypes.Head,
 }
 
 func (c *chainClient) CheckTxValidity(ctx context.Context, from common.Address, to common.Address, data []byte) *SendError {
-	msg := ethereum.CallMsg{
-		From: from,
-		To:   &to,
-		Data: data,
+	reqs := []TxSimulationRequest{
+		{
+			From: from,
+			To:   &to,
+			Data: data,
+		},
+	}
+	err := c.BatchCheckTxValidity(ctx, reqs)
+	if err != nil {
+		return NewSendError(err)
```

Comment:
`BatchCheckTxValidity` can fail for multiple reasons, for example the multi node might not be able to select a proper active node. That fail will not be related to the error classification. Perhaps it makes sense to return two values, SendError & error so the callers can tell if the call succeeded or not.

Reply:
I was considering that originally but was reluctant to change the method signature product teams are currently using. The main reason being product teams aren't doing more than just a nil check on the error which they can do to the SendError as well. Guess that's an unnecessary limitation on the error handling though so I can split this into 2 error.

Reply:
Pushed the change in the following commit